File: letterChanges.rb
require 'pry'

def LetterChanges(str)
  split_str = str.split(" ")
  binding.pry
  message = []
  letters = ["a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n", "o", "p", "q", "r", "s", "t", "u", "v", "w", "x", "y","z", "a"]
  vowels = ["a", "e", "i", "o", "u"]
  split_str.each do |str_letter|
    current_index = letters.index(str_letter)
    current_index += 1
    next_letter = letters[current_index]
    if vowels.include? next_letter
      next_letter.capitalize!
    end
    message << next_letter
  end
  message.join
  return message      
end

LetterChanges("Markymark and the funky bunch")

File: letterchanges.js
function LetterChanges(str) { 
  var alpha = ["a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n", "o", "p", "q", "r", "s", "t", "u", "v", "w", "x", "y", "z", "a"];
  var vowels = ["a", "e", "i", "o", "u"];
  letters = str.split('');
  newArray = []
  for (x=0; x<=letters.length; x++){
    index = alpha.indexOf(letters[x]);
    if (index == -1) {
      newArray.push(letters[x]);
    } else {
      var newLetter = alpha[index+1];
      if (vowels.indexOf(newLetter) == -1) {
      	newArray.push(newLetter);
      } else {
        newArray.push(newLetter.toUpperCase());
      }
    }
  }
  return newArray.join('');          
}
   
LetterChanges(readline());    